host/xtest/regression_1000.c

```diff
@@ -2687,3 +2694,358 @@ static void xtest_tee_test_1036(ADBG_Case_t *c)
+
+/*
+ * buf = [ TA hash (32 bytes SHA256) | Signature (att_key_size bytes) ]
+ * Signature = RSA_SSA_PKCS1_PSS_MGF1(SHA256([nonce | TA hash])
+ */
+static void check_signature(ADBG_Case_t *c, uint8_t *nonce, size_t nonce_size,
+			    uint8_t *buf)
+{
+	unsigned char digest[SHA256_DIGEST_LENGTH] = { };
+	unsigned char *sig = buf + SHA256_DIGEST_LENGTH;
+	unsigned char *ta_hash = buf;
+	uint8_t decr[ATT_MAX_KEYSZ / 8] = { };
+	SHA256_CTX ctx = { };
+	int salt_len = 32; /* Hard-coded in the PTA */
+	int st = 0;
+
+	ADBG_EXPECT_NOT_NULL(c, att_key);
+	if (!att_key)
+		return;
+
+	SHA256_Init(&ctx);
+	SHA256_Update(&ctx, nonce, nonce_size);
+	SHA256_Update(&ctx, ta_hash, SHA256_DIGEST_LENGTH);
+	SHA256_Final(digest, &ctx);
+
+	st = RSA_public_decrypt(att_key_size, sig, decr, att_key,
+				RSA_NO_PADDING);
+	ADBG_EXPECT_COMPARE_SIGNED(c, st, >, 0);
+	st = RSA_verify_PKCS1_PSS_mgf1(att_key, digest, EVP_sha256(),
+				       EVP_sha256(), decr, salt_len);
+	ADBG_EXPECT_COMPARE_SIGNED(c, st, >, 0);
+}
+
+static void free_att_key(void)
+{
+	RSA_free(att_key);
+	att_key = NULL;
+}
+
+static void set_att_key(ADBG_Case_t *c, uint8_t *e, size_t e_sz, uint8_t *n,
+			size_t n_sz)
+{
+	BIGNUM *bn_e = NULL;
+	BIGNUM *bn_n = NULL;
+	int st = 0;
+
+	att_key_size = n_sz;
+	att_key = RSA_new();
+	ADBG_EXPECT_NOT_NULL(c, att_key);
+
+	bn_e = BN_bin2bn(e, e_sz, NULL);
+	ADBG_EXPECT_NOT_NULL(c, bn_e);
+	bn_n = BN_bin2bn(n, n_sz, NULL);
+	ADBG_EXPECT_NOT_NULL(c, bn_n);
+
+	st = RSA_set0_key(att_key, bn_n, bn_e, BN_new());
+	ADBG_EXPECT_COMPARE_SIGNED(c, st, !=, 0);
+	if (!st)
+		free_att_key();
+}
+#else
+#define check_signature(...)
+#define set_att_key(...)
+#define free_att_key()
+#endif
+
+/*
+ * Verification of the output of the attestation PTA
+ * - (If hash != NULL) check buf contains the expected hash
+ * - (If OpenSSL is available) Check that the signature is valid
+ */
+static void check_measurement(ADBG_Case_t *c, uint8_t *nonce, size_t nonce_size,
+			      uint8_t *hash, uint8_t *buf)
+{
+	if (hash)
+		ADBG_EXPECT_BUFFER(c, hash, 32, buf, 32);
+
+	check_signature(c, nonce, nonce_size, buf);
+}
+
+/* Invoke attestation PTA to return the public key */
+static void get_att_public_key(ADBG_Case_t *c)
+{
+	uint8_t n[ATT_MAX_KEYSZ / 8] = { 0 };
+	uint8_t e[3] = { 0 };  /* We know e == 65537... */
+	size_t n_sz = sizeof(n);
+	size_t e_sz = sizeof(e);
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	TEEC_UUID att_uuid = PTA_ATTESTATION_UUID;
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Session session = { 0 };
+	uint32_t ret_orig = 0;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_VALUE_OUTPUT, TEEC_NONE);
+	op.params[0].tmpref.buffer = e;
+	op.params[0].tmpref.size = e_sz;
+	op.params[1].tmpref.buffer = n;
+	op.params[1].tmpref.size = n_sz;
+
+	res = xtest_teec_open_session(&session, &att_uuid, NULL, &ret_orig);
+	ADBG_EXPECT_TEEC_SUCCESS(c, res);
```

Comment:
This could also be written as:
``` C
if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
          return;
```
There's not really a need to check `res` again.